vita3k/mem/src/mem.cpp

```diff
@@ -227,54 +230,47 @@ void unprotect_inner(MemState &state, Address addr, size_t size) {
 #ifdef WIN32
     DWORD old_protect = 0;
     const BOOL ret = VirtualProtect(&addr_ptr[addr], size - 1, PAGE_READWRITE, &old_protect);
-    LOG_CRITICAL_IF(!ret, "VirtualAlloc failed: {}", log_hex(GetLastError()));
+    LOG_CRITICAL_IF(!ret, "VirtualAlloc failed: {}", get_error_msg());
 #else
-    mprotect(&addr_ptr[addr], size, PROT_READ | PROT_WRITE);
+    const int ret = mprotect(&addr_ptr[addr], size, PROT_READ | PROT_WRITE);
+    LOG_CRITICAL_IF(ret == -1, "mprotect failed: {}", get_error_msg());
 #endif
 }
 
-void protect_inner(MemState &state, Address addr, size_t size, const std::uint32_t perm) {
+void protect_inner(MemState &state, Address addr, uint32_t size, const uint32_t perm) {
     uint8_t *addr_ptr = state.use_page_table ? state.page_table[addr / KiB(4)] : state.memory.get();
 
 #ifdef WIN32
     DWORD old_protect = 0;
     const BOOL ret = VirtualProtect(&addr_ptr[addr], size - 1, (perm == MEM_PERM_NONE) ? PAGE_NOACCESS : ((perm == MEM_PERM_READONLY) ? PAGE_READONLY : PAGE_READWRITE), &old_protect);
```

Comment:
It could be good to make MEM_PERM_* enum named and use its type instead of uint32_t

Reply:
done